core/os/android/adb/commands.go

```diff
@@ -41,7 +41,7 @@ const (
 	ErrRootFailed      = fault.Const("Device failed to switch to root")
 	ErrUnrootFailed    = fault.Const("Device failed to switch out of root")
 
-	maxRootAttempts                         = 5
+	maxRootAttempts                         = 50
```

Comment:
I think we should increase the sleep time in response to the "restarting as ..." message, rather than the number of tries. I suspect that what is happening is that each successive time we run `adb [un]root` it is too early and adb aborts whatever work it was doing from the previous call and increasing the number of tries just means you increase your chances of getting lucky that it is fast enough.

In situations where the current version fails, how many times did it need to retry to succeed?

Reply:
purvisa-macbookpro3:~ purvisa$ adb root
adbd is already running as root
purvisa-macbookpro3:~ purvisa$ adb unroot
restarting adbd as non root
purvisa-macbookpro3:~ purvisa$ adb unroot
adbd not running as root
purvisa-macbookpro3:~ purvisa$ adb unroot
adbd not running as root
purvisa-macbookpro3:~ purvisa$ adb root # notice back of output from this command
purvisa-macbookpro3:~ purvisa$ adb root
adbd is already running as root

Somtimes no message is printed at all. In this case our code spins 5 times without a sleep. This is going to cause 5 rapid polls and then a device failure. I've added a sleep to prevent this, so unexpected output or no output at all gets a sleep before trying again with another poll.

I also increased the time from 0.5s (which seems very short given how long manual testing has shown the server to restart on my macbook. The benefit of polling rapidly many times is we get a fast turn around when the server is quick to restart. Polling fewer times with longer waits will result in worse best case performance.